### DISCUSSION TITLE
Need help: Pico-W + PCM5102A

### DISCUSSION THREAD

**Discussion by Sigh666** [micropython/micropython#13984] [RP2040 / Pico]
Hi everyone,

I am new to GitHub. After completing smaller projects in the past, I became addicted to microprocessors and small computers such as the Pi. Unfortunately, I haven't learned how to code because that's just not something you typically do when your profession is being a mechanical engineer. However, I am teaching myself and have successfully built a weather station as well as a WLED-Module. Even though I made it through those projects, I am still very dependent on the help of this community or other communities.

My new project is going to be a Wi-Fi Speaker for multiroom audio playback. Unfortunately, I can't find a lot of information when working with the combination of a Pico-W and a PCM5102A, which is the reason for my following question.

My goal is to have one Raspberry Pi running to which I can connect my phone via Wi-Fi and transfer the music from my phone. The Pi should then transfer the music to my Pico-W's in multiple rooms via Wi-Fi. Connected to the Pico-W is a PCM5102A DAC board.

It is the first time I am working with a Pico-W, and I am facing the problem that I don't know how to set up the connection/code between the Pico-W and the PCM5102A, as well as how to configure it with my phone as described before. I tried to have a quick look at the Logitech Media Server, but couldn't get it running on my computer, as well as Volumio. I tried starting Volumio from a USB stick with my Windows computer, but it didn't work because the image couldn't be verified.

I would need help finding some kind of manual and code to put on my Pico-W. I already tried to find a solution for this project on GitHub without success. I would be very grateful for any help you could offer. So far, I have the Pico-W and PCM5102A wired up on a breadboard, but I'm stuck on the next steps.

Thanks again, and I'm looking forward to getting help from you.

Best regards,
Chris

![Photo 2024-03-01, 9 24 00 PM](https://github.com/micropython/micropython/assets/161881351/b28bb078-27f5-4601-b4e4-09eabb806375)
![Photo 2024-03-01, 9 24 12 PM](https://github.com/micropython/micropython/assets/161881351/d8fb59f8-e519-477b-8c15-2f91e5d4026f)
![PCM5102A](https://github.com/micropython/micropython/assets/161881351/0bd00831-00a1-4963-8d8e-769af85f4925)

**Comment by sophiedegran** [comment]
Hi
Have a look to this project
github.com/miketeachman/micropython-i2s-examples
It's not exactly what you want but i think it's good for starting
Hope that help
